multinet/db.py

```diff
@@ -132,358 +107,33 @@ def workspace_mapping(name: str) -> Optional[Dict]:
-    validate_csv(rows, "_key", False)
-
-    db = get_workspace_db(workspace, readonly=False)
-    coll = db.create_collection(name, sync=True)
-    coll.insert_many(rows)
-
-    return name
-
-
-def graph_node(workspace: str, graph: str, table: str, node: str) -> dict:
-    """Return the data associated with a particular node in a graph."""
-    space = get_workspace_db(workspace)
-    graphs = filter(lambda g: g["name"] == graph, space.graphs())
-    try:
-        next(graphs)
-    except StopIteration:
-        raise GraphNotFound(workspace, graph)
-
-    tables = filter(lambda t: t["name"] == table, space.collections())
-    try:
-        next(tables)
-    except StopIteration:
-        raise TableNotFound(workspace, table)
-
-    query = f"""
-    FOR d in {table}
-      FILTER d._id == "{table}/{node}"
-      RETURN d
-    """
-
-    result = aql_query(workspace, query)
-    try:
-        data = next(result)
-    except StopIteration:
-        raise NodeNotFound(table, node)
-
-    return {k: data[k] for k in data if k != "_rev"}
-
-
-def workspace_graphs(workspace: str) -> List[str]:
-    """Return a list of all graph names in workspace `workspace`."""
-    space = get_workspace_db(workspace)
-    return [graph["name"] for graph in space.graphs()]
+def user_collection() -> StandardCollection:
+    """Return the collection that contains user documents."""
+    sysdb = system_db(readonly=False)
 
+    if not sysdb.has_collection("users"):
+        sysdb.create_collection("users")
 
-def workspace_graph(workspace: str, graph: str) -> GraphSpec:
-    """Return a specific graph named `name` in workspace `workspace`."""
-    get_graph_collection(workspace, graph)
+    return sysdb.collection("users")
 
-    # Get the lists of node and edge tables.
-    node_tables = graph_node_tables(workspace, graph)
-    edge_table = graph_edge_table(workspace, graph)
 
-    return {"nodeTables": node_tables, "edgeTable": edge_table}
+def search_user(query: str) -> Cursor:
+    """Search for users given a partial string."""
 
+    coll = user_collection()
+    aql = system_db().aql
 
-def graph_nodes(workspace: str, graph: str, offset: int, limit: int) -> GraphNodesSpec:
-    """Return the nodes of a graph."""
-    get_graph_collection(workspace, graph)
+    bind_vars = {"@users": coll.name, "query": query}
+    query = """
+        FOR doc in @@users
+          FILTER CONTAINS(LOWER(doc.name), LOWER(@query))
+            OR CONTAINS(LOWER(doc.email), LOWER(@query))
 
-    # Get the actual node data.
-    node_tables = graph_node_tables(workspace, graph)
-    node_query = f"""
-    FOR c in [{", ".join(node_tables)}]
-      FOR d in c
-        LIMIT {offset}, {limit}
-        RETURN d
-    """
-    nodes = aql_query(workspace, node_query)
-
-    # Get the total node count.
-    count_query = f"""
-    FOR c in [{", ".join(node_tables)}]
-      FOR d in c
-        COLLECT WITH COUNT INTO count
-        RETURN count
+          LIMIT 50
```

Comment:
I know it has nothing to do with this PR, but let's file an issue to make this search function pageable.

Reply:
#462